### ISSUE TITLE
A11y Code Sprint Berlin: Success Criterion 9.1.3.1h - Info and Relationships - Forms

### ISSUE THREAD

**Issue by alexakressmann** [benjaminkott/bootstrap_package#1325]
**Page**
-
https://bitv-v12.live.typo3.cps.321.works/content-examples/form-elements/login

**Description**
-
Forms must offer `for `and `id`.
[To WCAG](https://www.w3.org/TR/WCAG21/#info-and-relationships)

**Solution**
-
Add `for `and `id` for the input fields like on the page [Search](https://bitv-v12.live.typo3.cps.321.works/content-examples/form-elements/search)

**Comment by linawolf** [comment]
@alexakressmann in the login form the input is found within the label, which should make the context clear:
![image](https://github.com/benjaminkott/bootstrap_package/assets/48202465/fb1c128e-ec6b-4dc3-8b09-20abb4b7c560)

Do we need id and for attributes nevertheless?

**Comment by linawolf** [comment]
In the search we already use id and for attributes:

![image](https://github.com/benjaminkott/bootstrap_package/assets/48202465/d9d50ad4-b44b-41a6-805a-71f2e4380dec)

**Comment by alexakressmann** [comment]
> @alexakressmann in the login form the input is found within the label, which should make the context clear: ![image](https://user-images.githubusercontent.com/48202465/243646395-fb1c128e-ec6b-4dc3-8b09-20abb4b7c560.png)
> 
> Do we need id and for attributes nevertheless?

Yes, we need it. Thank you

**Comment by linawolf** [comment]
See https://forge.typo3.org/issues/100965 and https://review.typo3.org/c/Packages/TYPO3.CMS/+/79194

**Comment by linawolf** [comment]
@alexakressmann please review

**Comment by alexakressmann** [comment]
QA successful

**Comment by rr-it** [comment]
@a-wunnderbar please see https://www.w3.org/WAI/WCAG22/Techniques/failures/F68.html#tests

> ### Procedure
> 
> For all `input`, `textarea`, and `select` elements in the Web page (except inputs of type `hidden`, `submit`, `reset`, or `button`:
> 
> 1. Check that each element has a programmatically determined name _using one of the following ways_:
>    1. […]
>    2. […]
>    3. the text label is contained in a `label` element that is correctly associated to the respective `input` element via the label's `for` attribute (the `id` given as value in the `for` attribute matches the `id` of the input element).
>    4. the control is contained within a `label` element that also contains the label text.
>    5. […]
>    6. […]

So I'd say the following code should be 'QA successful' as of '1.4.' above:

```html
<label>
    Username
    <input type="text" …>
</label>
```

----

Also see https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label

> To explicitly associate a `<label>` element with an <input> element, you first need to add the id attribute to the <input> element. Next, you add the for attribute to the <label> element, where the value of for is the same as the id in the <input> element.

    Alternatively, you can nest the <input> directly inside the <label>, in which case the for and id attributes are not needed because the association is implicit:


We really get into trouble if we need to set unique `id` attributes everywhere. As soon as a form is used twice on the same page, the `id` isn't unique anymore.
The result of a non-unique `id` results in a *wrong label-association*.  A wrong label-association is far worse than not providing the `id` and `for` attributes to 'improve' an already working label-to-input association.
